### DISCUSSION TITLE
Is here query execution order?

### DISCUSSION THREAD

**Discussion by Kagayakashi** [webonyx/graphql-php#1247] [Q&A] [answered]
I have custom debug from backend that have output into graphql with query field:

`query { debug { ... }, otherField }`

Problem is that if `debug` field is before `otherField`, then debug will be returned empty.
Debug will be empty because its executing first and after debug will be executed other field with debug methods inside.

Is here here any option to make the debug field execution at last?

**Comment by spawnia** [comment, marked as answer]
This library executes fields in the order they were sent from the query, there is no way to change that. Perhaps `extensions` is a better fit for what you are trying to do.

**Reply by Kagayakashi** [reply]
If I understood correct that debug in this lib shows only when there is Exception.
So I created custom one.

The idea is that in backend config I can enable debug mode.
Debug have "senders" and one of them is GraphqlSender which send debug output to Graphql as field in Query root.
With this feature frontend can output some debug from backend for example in `console.log` or just show it somehow.

Query Example
```
query {
  dicts(status: 1) {
    id,
    name,
  },
  debug {
    message
  }
}
```

Response example
```
{
  dicts: [
    { id: 1, name: "name1" },
    { id: 2, name: "name2" }
  ],
  debug: [
    { message: "Selected 2 dicts with id: [1, 2]"}
  ]
}
```

Ofc I can just write in documentation of API that debug fields need to be at last.
But i was thinking that there is some way to set execution order.

**Reply by spawnia** [reply]
As i said, the only reliable way I can think of is to use extensions. The implementation is totally up to you, you just need to set `ExecutionResult::$extensions` before serializing and returning the result.
